crates/noirc_driver/src/lib.rs

```diff
@@ -212,10 +212,23 @@ impl Driver {
 
             let func_type = ContractFunctionType::new(func_type, func_meta.is_unconstrained);
 
-            Ok((function_name, ContractFunction { func_type, function }))
+            Ok((function_name, func_type, function))
         })?;
 
-        Ok(CompiledContract { name: contract.name, functions })
+        let converted_functions: Vec<ContractFunction> = functions
+            .into_iter()
+            .map(|(func_name, func_type, function)| ContractFunction {
+                name: func_name,
+                function_type: func_type,
+                abi: function.abi,
+                bytecode: function.circuit,
+                // Since we have not called the proving system yet
+                // we do not have a verification key
+                verification_key: None,
+            })
+            .collect();
```

Comment:
```suggestion
        let converted_functions = vec_map(
            functions,
            |(name, function_type, function)| ContractFunction {
                name,
                function_type,
                abi: function.abi,
                bytecode: function.circuit,
                // Since we have not called the proving system yet
                // we do not have a verification key
                verification_key: None,
            });
```

We can use a `vecmap` here and avoid some renamings